aioslimproto/discovery.py

```diff
@@ -199,15 +200,20 @@ def datagram_received(self, data, addr):
         """Datagram received callback."""
         # pylint: disable=broad-except
         try:
-            data = data.decode()
             dgram = Datagram.decode(data)
             if isinstance(dgram, ClientDiscoveryDatagram):
                 self.send_discovery_response(addr)
             elif isinstance(dgram, TLVDiscoveryRequestDatagram):
                 resonsedata = self.build_tlv_response(dgram.data)
                 self.send_tlv_discovery_response(resonsedata, addr)
         except Exception as exc:
-            LOGGER.exception(exc)
+            LOGGER.error(
+                "Error occured while trying to parse a datagram from %s: %s",
+                addr,
+                str(exc),
+                exc_info=exc,
+            )
+            LOGGER.error(data)
```

Comment:
Why not include the data in the above error log message?

Reply:
because I have no idea about the codec being used. I want to prevent the actual exception failed being logged.